### ISSUE TITLE
Chore: Clippy Code Cleanup Story Ticket

### ISSUE THREAD

**Issue by jferrant** [stacks-network/stacks-blockchain#3934]
This is a main story ticket for cleaning up the stacks-blockchain repo by addressing clippy errors throughout the entire code base.

To run clippy on just the offending package :
`cargo clippy -p PACKAGE_TO_CHECK --no-deps`

Subtasks include:

- [ ] clarity: address all clippy complaints
- [ ] libstackerdb: adddress all clippy complaints
- [ ] pox-locking: address all clippy complaints
- [ ] stacks-common: address all clippy complaints
- [ ] stackslib: address all clippy complaints
- [ ] stx-genesis: address all clippy complaints
- [ ] testnet: address all clippy complaints

**Comment by obycode** [comment]
I'd suggest we use `-- -D warnings` to treat the warnings as errors too.

**Comment by obycode** [comment]
Adding a note here that once `develop` is merged into `next`, including the clippy checks, we should enable those checks for PRs against `next` as well (it's currently only enabled for PRs against `develop`).
